### PR TITLE
raftstore: enable observing cmds for all regions

### PR DESCRIPTION
**Pull request by 5kbpers**
Signed-off-by: 5kbpers <tangminghua@pingcap.com>

### What problem does this PR solve?

Problem Summary:
Separate from #9435

After introducing follower stale read, all commands executed by all leaders need to be observed to advance resolved ts.

### What is changed and how it works?

What's Changed:
This PR temporarily enable to observe commands for all regions.
There will be a mechanism to disable observing for follower peers after we complete merging all codes related to resolved ts.

### Check List 

Tests 

- Unit test
- Integration test

### Release note 
* N/A

### CODE REVIEW

File: components/raftstore/src/store/fsm/apply.rs [hunk 1]
```diff
@@ -4755,19 +4710,18 @@ mod tests {
             .build();
         router.schedule_task(1, Msg::apply(apply(peer_id, 1, 2, vec![put_entry], vec![])));
         // Must not receive new cmd.
-        cmdbatch_rx
-            .recv_timeout(Duration::from_millis(100))
-            .unwrap_err();
+        // cmdbatch_rx
+        //     .recv_timeout(Duration::from_millis(100))
+        //     .unwrap_err();
```
Comment (overvenus) [inline components/raftstore/src/store/fsm/apply.rs:4715]: Need clean up?

File: components/raftstore/src/store/fsm/apply.rs [hunk 2]
```diff
@@ -2847,16 +2842,9 @@ impl Debug for ObserveCmd {
...
-    RegisterObserver {
-        observe_id: ObserveID,
-        region_id: u64,
-        enabled: Arc<AtomicBool>,
-    },
-    Snapshot {
-        observe_id: ObserveID,
-        region_id: u64,
-    },
+pub struct ChangeCmd {
```
Comment (overvenus) [inline components/raftstore/src/store/fsm/apply.rs:2845]: Should it be rename to a proper name? let's say "ChangeObserver"?

**Review by overvenus — COMMENTED** [review]
Rest LGTM

**Comment by NingLin-P** [comment]
Please address the comments above, thanks!

**Pull request merged**